modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/search/spi/model/index/contributor/CTEntryModelDocumentContributor.java

```diff
@@ -180,6 +183,19 @@ private <T extends BaseModel<T>> void _indexModelAttributes(
 			}
 		}
 
+		if (model instanceof PortletPreferences) {
+			PortletPreferences portletPreferences = (PortletPreferences)model;
+
+			Layout layout = _layoutLocalService.fetchLayout(
+				portletPreferences.getPlid());
+
+			if (layout != null) {
+				document.addKeyword(Field.GROUP_ID, layout.getGroupId());
+				document.addLocalizedKeyword(
+					"groupName", layout.getNameMap(), true, true);
+			}
+		}
+
```

Comment:
Let's generalize this so we also cover other entities that have a plid. Can you add another condition in https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/search/spi/model/index/contributor/CTEntryModelDocumentContributor.java#L191-L195 to get the groupId from plid?